lib/content-services/src/lib/search/components/search-date-range-advanced-tabbed/search-date-range-advanced/search-date-range-advanced.component.ts

```diff
@@ -0,0 +1,163 @@
+    providers: [
+        { provide: DateAdapter, useClass: DateFnsAdapter, deps: [ MAT_DATE_LOCALE ] },
+        { provide: MAT_DATE_FORMATS, useValue: MAT_DATE_FNS_FORMATS }
+    ],
+    encapsulation: ViewEncapsulation.None,
+    host: {class: 'adf-search-date-range-advanced'}
+})
+export class SearchDateRangeAdvancedComponent implements OnInit, OnDestroy {
+    @Input()
+    dateFormat = DEFAULT_DATE_DISPLAY_FORMAT;
+    @Input()
+    maxDate: string;
+    @Input()
+    field: string;
+    @Input()
+    set initialValue(value: SearchDateRangeAdvanced) {
+        if (value) {
+            this.form.patchValue(value);
+        }
+    }
+
+    @Output()
+    changed = new EventEmitter<Partial<SearchDateRangeAdvanced>>();
+    @Output()
+    valid = new EventEmitter<boolean>();
+
+    form = this.formBuilder.group<SearchDateRangeAdvanced>({
+        dateRangeType: DateRangeType.ANY,
+        inLastValueType: InLastDateType.DAYS,
+        inLastValue: undefined,
+        betweenStartDate: undefined,
+        betweenEndDate: undefined
+    });
+    betweenStartDateFormControl = this.form.controls.betweenStartDate;
+    betweenEndDateFormControl = this.form.controls.betweenEndDate;
+    convertedMaxDate: Date;
+    private destroy$ = new Subject<void>();
+
+    readonly DateRangeType = DateRangeType;
+    readonly InLastDateType = InLastDateType;
+
+    constructor(private formBuilder: FormBuilder,
+                private userPreferencesService: UserPreferencesService,
+                private dateAdapter: DateAdapter<DateFnsAdapter>,
+                @Inject(MAT_DATE_FORMATS) private dateFormatConfig:MatDateFormats) {}
+
+    readonly endDateValidator = (formControl: UntypedFormControl): ({ [key: string]: boolean } | null) => {
+        if (isBefore(formControl.value, this.betweenStartDateFormControl.value) || isAfter(formControl.value, this.convertedMaxDate)) {
+            return {
+                invalidDate: true
+            };
+        }
+        return {};
+    }
+
+    ngOnInit(): void {
+        this.dateFormatConfig.display.dateInput = this.dateFormat;
+        this.convertedMaxDate = endOfDay(this.maxDate && this.maxDate !== 'today' ?
+            parse(this.maxDate, this.dateFormat, new Date()) : new Date());
+        this.userPreferencesService
+            .select(UserPreferenceValues.Locale)
+            .pipe(takeUntil(this.destroy$))
+            .subscribe(locale => this.dateAdapter.setLocale(DateFnsUtils.getLocaleFromString(locale)));
+        this.form.controls.dateRangeType.valueChanges.pipe(takeUntil(this.destroy$))
+            .subscribe((dateRangeType) => this.updateValidators(dateRangeType));
+        this.form.valueChanges.pipe(takeUntil(this.destroy$))
+            .subscribe(() => this.onChange());
+    }
+
+    ngOnDestroy() {
+        this.destroy$.next();
+        this.destroy$.complete();
+    }
+
+    private updateValidators(dateRangeType: DateRangeType) {
+        switch(dateRangeType) {
```

Comment:
eslint issue: 
<img width="841" alt="image" src="https://github.com/Alfresco/alfresco-ng2-components/assets/115619721/9f029178-fc81-499c-ba88-366f94347bd9">

Reply:
These has been resolved in [this](https://github.com/Alfresco/alfresco-ng2-components/pull/8714/commits/bdaa0e52fd3a31ba7c3802c6cdc6a27a733d6057) commit